### ISSUE TITLE
Art: Potions

### ISSUE THREAD

**Issue by madjin** [webaverse/app#1347]
Potions will buff/debuff players to have extreme experiences in-game. I think what would go a long way is to think about the types of potions we will have, what ingredients they will require, and perhaps how the icons for those ingredients / potions can gracefully degrade into tiny icons for 2D menus and Discord.

![image](https://user-images.githubusercontent.com/32600939/124814701-56a6d680-df34-11eb-8c9b-5956f78c47e6.png)

![image](https://user-images.githubusercontent.com/32600939/124814720-5c042100-df34-11eb-9275-1114b712c380.png)

![image](https://user-images.githubusercontent.com/32600939/124814899-8e158300-df34-11eb-961b-c80b58180692.png)

**Comment by avaer** [comment]
This should be its own configurable NFT type in the editor -- upload the model, with some way to animate the "fill" state, and a way to input effects.

**Comment by plankatron** [comment]
![potion1](https://user-images.githubusercontent.com/5592454/124989279-fcc60f80-e00c-11eb-9f85-a33f7fa09cf6.png)

**Comment by avaer** [comment]
The thing I had in mind was something slurpable, like the jar in the first reference:

![](https://user-images.githubusercontent.com/32600939/124814701-56a6d680-df34-11eb-8c9b-5956f78c47e6.png)

The last reference set (which did not come from the docs) is the most traditional/forgettable option and it seems that is what these models are based off of.

But the star of the show is less the containers and more the contents.

We should get a base texture (or set of base textures) for the contents, which could be animated to swirl or slosh.

**Comment by avaer** [comment]
Also I think we only need one container to start, with changeable content textures.

**Comment by avaer** [comment]
For future iterations and/or texture inspiration:

![Collection of Potion Bottles for Prop Concepts](https://user-images.githubusercontent.com/6926057/125000278-0f951000-e01e-11eb-8b97-cb8340fdedc2.png)

**Comment by plankatron** [comment]
https://user-images.githubusercontent.com/5592454/125010445-85a47180-e034-11eb-8942-b9eb96896c92.mp4

**Comment by avaer** [comment]
That looks sick! Though a cylinder is probably a better shape for avatar holding purposes (hopefully that’s not a difficult change).

For the materials I think the above shot has a good variety of textures and colors. I’m even down for adding particles floating around. As long as the animations and textures are baked and the content objects are separate so they can be enabled and disabled. There is a lot of leeway.

The only potion that appears in the trailer is a speed one and I don’t even think it matters what it looks like — the above is good enough if it can be baked.

**Comment by avaer** [comment]
Re transparency, I’m not so much concerned about the reflections and stuff. Even going down to Zelda bottle mode would not be terrible.

**Comment by plankatron** [comment]
another test , tomorrow  i will experiment with some glb exports 
![jar1](https://user-images.githubusercontent.com/5592454/125027247-f0b17080-e053-11eb-9715-95c3de2b4c5b.jpg)

**Comment by avaer** [comment]
That looks great, but I meant rectangularizing the jar, not replacing it with the zelda bottle form:

![image](https://user-images.githubusercontent.com/6926057/125029646-ff9a2200-e057-11eb-8778-e32f171005d8.png)

I was only referencing the zelda bottle as a possible way to make the transparency work cleanly.

However it seems both of the ingredients are here, they just need to be combined.

**Comment by Vianvolaeus** [comment]
https://pathofexile.fandom.com/wiki/List_of_unique_flasks

Just dropping this in here, some of these are great in terms of tying the IPs (dark fantasy) theme with gameplay and design:

![image](https://user-images.githubusercontent.com/80061436/125085515-913e7980-e0c2-11eb-87d9-8bb0b47c6ae8.png)

These are looking amazing, though

**Comment by plankatron** [comment]
like this avaer: ? 
![Screenshot at 2021-07-09 17-20-51](https://user-images.githubusercontent.com/5592454/125137133-15502b80-e0da-11eb-8451-b7deb20bde5d.png)

**Comment by plankatron** [comment]
variant : 
![Screenshot at 2021-07-09 18-01-14](https://user-images.githubusercontent.com/5592454/125140182-edfc5d00-e0df-11eb-8d88-ece30d6ef48b.png)

**Comment by plankatron** [comment]
so this is a pretty good test case , but i think its a bit too generic, can you post specific screens of potion direction?
![Screenshot at 2021-07-09 18-10-06](https://user-images.githubusercontent.com/5592454/125140709-22244d80-e0e1-11eb-934e-8fea1d89ab36.png)

**Comment by avaer** [comment]
> https://github.com/webaverse/app/issues/1347#issuecomment-877462878

Yes this version works. I see what you mean by not rendering back faces, but we can fix that and it is not a major issue.

**Comment by avaer** [comment]
This is marked as in review; is that asset up somewhere?

**Comment by madjin** [comment]
@plankatron had to fix the animation on the glb's so they loop correctly, will be up soon

**Comment by plankatron** [comment]
animated and static potions are on gdrive please review

**Comment by madjin** [comment]
Nice renders + transparent images
![round_jar_transp](https://user-images.githubusercontent.com/32600939/125525779-fa22719e-d29e-42bb-a4ac-5e55199493a6.png)
![square_jar_transp](https://user-images.githubusercontent.com/32600939/125525787-c944ef88-8488-4018-a2e7-e8a71758bee0.png)

![image](https://user-images.githubusercontent.com/32600939/125525309-725a9642-f1e2-49f6-9261-00b42b4daa2a.png)


Verified that the animated potions loop perfectly in sandbox / gltf viewer. 

Checked the static ones too. Great job on getting a really low filesize, these can make great scene props and templates for future artists.

What repo are we going to upload these to?

**Comment by avaer** [comment]
> What repo are we going to upload these to?

https://github.com/webaverse/assets

**Comment by madjin** [comment]
added
